libgalois/include/katana/Properties.h

```diff
@@ -182,6 +171,106 @@ ConstructPropertyViews(const std::vector<arrow::Array*>& arrays) {
       arrays, std::make_index_sequence<std::tuple_size_v<PropTuple>>());
 }
 
+template <typename T>
+struct HasAllocate {
+  template <typename U>
+  constexpr static int test(decltype(&U::Allocate)) {
+    return true;
+  }
+
+  template <typename>
+  constexpr static bool test(...) {
+    return false;
+  }
+
+  constexpr static bool value = sizeof(test<T>(nullptr)) == sizeof(int);
+};
+
+template <typename Prop>
+std::enable_if_t<
+    !HasAllocate<Prop>::value, Result<std::shared_ptr<arrow::Table>>>
+Allocate(size_t num_rows, const std::string& name) {
+  using ArrowType = typename PropertyTraits<Prop>::ArrowType;
+  using Builder = typename arrow::TypeTraits<ArrowType>::BuilderType;
+  using CType = typename arrow::TypeTraits<ArrowType>::CType;
+  Builder builder;
+
+  std::vector<CType> rows(num_rows);
+  if (auto r = builder.AppendValues(rows); !r.ok()) {
```

Comment:
Where is `rows` defined?

Reply:
I don't get this comment. It is defined at line 198

Reply:
Never mind. Thanks.